pkg/issuer/acme/dns/util/wait.go

```diff
@@ -78,6 +99,33 @@ func checkDNSPropagation(fqdn, value string) (bool, error) {
 	return checkAuthoritativeNss(fqdn, value, authoritativeNss)
 }
 
+func checkDNSPropagationWithHTTPS(fqdn, value string) (bool, error) {
+	r, err := http.Get("https://cloudflare-dns.com/dns-query?ct=application/dns-json&type=TXT&name=" + fqdn)
```

Comment:
If outbound DNS connectivity is as restricted as you're talking about, wouldn't this also fail?

Reply:
By definition we have access to DNS provider API and therefore this lookup is also possible.  This case can happen when cert-manager runs behind HTTP proxy with no way around it.